source/math/FloatMatrix.ooc

```diff
@@ -395,6 +365,28 @@ FloatMatrix: cover {
 		this _elements = this _elements give()
 		this
 	}
+	// NOTE: Because rock doesn't understand the concept of inline functions,
+	// this function has been inlined manually in many places in this file for performance reasons.
+	operator [] (x, y: Int) -> Float {
+		t := this take()
+		version (safe) {
+			if (x < 0 || y < 0 || x >= t width || y >= t height)
+				raise("Accessing matrix element out of range in get operator")
+		}
+		result := t elements[x + y * t width]
+		this free(Owner Receiver)
+		result
+	}
+	// NOTE: Because rock doesn't understand the concept of inline functions,
+	// this function has been inlined manually in many places in this file for performance reasons.
```

Comment:
We do this in a lot of places these days, so you can probably remove these comments too.